File: 0278-first-bad-version/solution.cpp
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        
        long long int min=0,max=n;
        long int i;
        while(min<=max){
            i = (max+min)/2; //i =1
            if(isBadVersion(i)){
                max = i-1; // 2
            }
            else min=i+1;
        }
        return min;
    }
};

File: 0278-first-bad-version/solution.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        
        min_ = 0
        max_ = n
        
        while True:
          mid = int((min_+max_)/2)
          
          if isBadVersion(mid):
            max_ = mid - 1
            
            if isBadVersion(mid-1) == False:
              return mid  
          else:
              min_ = mid + 1
          
        return n
            
          
          
